### PR TITLE
chore: Update notifications engine to 7a06976

### PR DESCRIPTION
**Pull request by kzap**
Fixes https://github.com/argoproj/argo-rollouts/issues/3049

## What
1. Updates in this latest version https://github.com/argoproj/notifications-engine/commit/7a069766e95476e1074eeed6145085160e2fec63 of `notifications-engine`:
  - https://github.com/argoproj/notifications-engine/pull/259
  - **SQS**:
    - https://github.com/argoproj/notifications-engine/pull/261
  - **GitHub**:
    - https://github.com/argoproj/notifications-engine/pull/260
    - https://github.com/argoproj/notifications-engine/pull/235
  - **Slack**:
    - https://github.com/argoproj/notifications-engine/pull/242
  - **Opsgenie**:
    - https://github.com/argoproj/notifications-engine/pull/267
    - https://github.com/argoproj/notifications-engine/pull/268
  - **Telegram**:
    - https://github.com/argoproj/notifications-engine/pull/265

2. Updating `notifications` documentation generator to replace some `argocd-` prefixed strings.

3. Adding documentation for contributors on how to update `notifications-engine` like in this PR.

## Checklist

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/) with a list of types and scopes found [here](https://github.com/argoproj/argo-rollouts/blob/master/.github/workflows/pr-title-check.yml), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -202,6 +202,7 @@ require (
 
 replace (
 	github.com/go-check/check => github.com/go-check/check v0.0.0-20180628173108-788fd7840127
+	github.com/go-telegram-bot-api/telegram-bot-api/v5 => github.com/OvyFlash/telegram-bot-api/v5 v5.0.0-20240108230938-63e5c59035bf
```
Comment (kzap) [inline go.mod:205]: replaced in the notifications engine in this PR: https://github.com/argoproj/notifications-engine/pull/265

File: go.mod [hunk 2]
```diff
@@ -4,7 +4,7 @@ go 1.20
 
 require (
 	github.com/antonmedv/expr v1.15.5
-	github.com/argoproj/notifications-engine v0.4.1-0.20231213155711-c02dc5fdb5e1
+	github.com/argoproj/notifications-engine v0.4.1-0.20240219110818-7a069766e954
```
Comment (kzap) [inline go.mod:7]: Updating to https://github.com/argoproj/notifications-engine/commit/7a069766e95476e1074eeed6145085160e2fec63

File: hack/gen-docs/main.go [hunk 3]
```diff
@@ -36,6 +36,12 @@ func generateNotificationsDocs() {
 		if e := updateMkDocsNav("Notifications", "Services", files); e != nil {
 			log.Fatal(e)
 		}
+		if e := strReplaceDocFiles("argocd-notifications-cm", "argo-rollouts-notification-configmap", files); e != nil {
+			log.Fatal(e)
+		}
+		if e := strReplaceDocFiles("argocd-notifications-secret", "argo-rollouts-notification-secret", files); e != nil {
+			log.Fatal(e)
+		}
```
Comment (kzap) [inline hack/gen-docs/main.go:44]: lets replace these strings with the `argo-rollouts-` equivalent

File: docs/generated/notification-services/alertmanager.md [hunk 4]
```diff
@@ -43,7 +43,7 @@ You should turn off "send_resolved" or you will receive unnecessary recovery not
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: <config-map-name>
+  name: argo-rollouts-notification-configmap
```
Comment (kzap) [inline docs/generated/notification-services/alertmanager.md:46]: generated by running `make docs`

These were recently renamed upstream to a `argocd` specific name :/

**Review by zachaller — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**